### PR TITLE
network labels

### PR DESCRIPTION
**Pull request by dspeck1**
Add add in network label support.   Encountered Google library issue so had to bump versions to resolve error with load balancing.

### CODE REVIEW

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala [hunk 1]
```diff
@@ -110,6 +120,82 @@ class GcpBatchInitializationActor(batchParams: GcpBatchInitializationActorParams
...
   }
 
+  private def vpcNetworkAndSubnetworkProjectLabelsFuture(): Future[Option[VpcAndSubnetworkProjectLabelValues]] = {
+    def googleProject(descriptor: BackendWorkflowDescriptor): String = {
+      descriptor.workflowOptions.getOrElse(WorkflowOptionKeys.GoogleProject, batchParams.batchConfiguration.batchAttributes.project)
+    }
+
+    def projectMetadataRequest(vpcConfig: VirtualPrivateCloudLabels): Future[HttpRequest] = {
+      Future {
+        val credentials = vpcConfig.auth.credentials(workflowOptions.get(_).get, List(CloudLifeSciencesScopes.CLOUD_PLATFORM))
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala]: `workflowOptions.get(_).get` seems to be `Option#get` which is discouraged due to the lack of a proper error message, an alternative for this case could be `workflowOptions.get(_).getOrElse(throw new RuntimeException("Unable to find the necessary workflow option))`, ideally, add more context to the error message

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala [hunk 2]
```diff
@@ -110,6 +120,82 @@ class GcpBatchInitializationActor(batchParams: GcpBatchInitializationActorParams
...
+        val project = cloudResourceManagerBuilder.projects().get(googleProject(workflowDescriptor))
+
+        project.buildHttpRequest()
+      }
+    }
+
+    def projectMetadataResponseToLabels(httpResponse: HttpResponse): Future[ProjectLabels] = {
+      implicit val googleProjectMetadataLabelDecoder: Decoder[ProjectLabels] = deriveDecoder
+      Future.fromTry(decode[ProjectLabels](httpResponse.parseAsString()).toTry).recoverWith {
+        case e: Throwable => Future.failed(new RuntimeException(s"Failed to parse labels from project metadata response from Google Cloud Resource Manager API. " +
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala]: Usually, catching `Throwable` is not recommended, instead do `case NonFatal(e) => `

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/ProjectLabels.scala [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package cromwell.backend.google.pipelines.batch
```
Comment (AlexITC) [inline supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/ProjectLabels.scala]: This package has already many classes, I'd suggest creating `cromwell.backend.google.pipelines.batch.models` instead

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala (hunk 1 above)
Comment (dspeck1) [reply supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala]: Thanks!  Adjusted.

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala (hunk 2 above)
Comment (dspeck1) [reply supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/GcpBatchInitializationActor.scala]: Thanks!  Adjusted.

File: supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/ProjectLabels.scala (hunk 3 above)
Comment (dspeck1) [reply supportedBackends/google/pipelines/batch/src/main/scala/cromwell/backend/google/pipelines/batch/ProjectLabels.scala]: Thanks!  Adjusted.

**Pull request merged**